tests/shared/s3-packages.spec.js

```diff
@@ -0,0 +1,988 @@
+              <Size>72042525</Size>
+              <StorageClass>STANDARD</StorageClass>
+            </Contents>
+          </ListBucketResult>
+        `);
+      });
+
+      afterEach(function() {
+        nock.cleanAll();
+      });
+
+      it('should resolve an empty array', function(done) {
+        s3Packages.getRemoteVersions(s3Packages.BUCKET_URL.PRODUCTION).then((versions) => {
+          m.chai.expect(versions).to.deep.equal([]);
+          done();
+        }).catch(done);
+      });
+
+    });
+
+    describe('given an unsuccessful request', function() {
+
+      beforeEach(function() {
+        nock(s3Packages.BUCKET_URL.PRODUCTION).get('/').reply(500);
+      });
+
+      afterEach(function() {
+        nock.cleanAll();
+      });
+
+      it('should be rejected with an error', function(done) {
+        s3Packages.getRemoteVersions(s3Packages.BUCKET_URL.PRODUCTION).catch((error) => {
+          m.chai.expect(error).to.be.an.instanceof(Error);
+          done();
+        });
+      });
+
+    });
+
+    describe('given ENOTFOUND', function() {
+
+      beforeEach(function() {
+        const error = new Error('ENOTFOUND');
+        error.code = 'ENOTFOUND';
+
+        this.requestGetAsyncStub = m.sinon.stub(request, 'getAsync');
+        this.requestGetAsyncStub.returns(Bluebird.reject(error));
+      });
+
+      afterEach(function() {
+        this.requestGetAsyncStub.restore();
+      });
+
+      it('should return an empty array', function(done) {
+        s3Packages.getRemoteVersions(s3Packages.BUCKET_URL.PRODUCTION).then((versions) => {
+          m.chai.expect(versions).to.deep.equal([]);
+          done();
+        }).catch(done);
+      });
+
+    });
+
+    describe('given ETIMEDOUT', function() {
+
+      beforeEach(function() {
+        const error = new Error('ETIMEDOUT');
+        error.code = 'ETIMEDOUT';
+
+        this.requestGetAsyncStub = m.sinon.stub(request, 'getAsync');
+        this.requestGetAsyncStub.returns(Bluebird.reject(error));
+      });
+
+      afterEach(function() {
+        this.requestGetAsyncStub.restore();
+      });
+
+      it('should return an empty array', function(done) {
+        s3Packages.getRemoteVersions(s3Packages.BUCKET_URL.PRODUCTION).then((versions) => {
+          m.chai.expect(versions).to.deep.equal([]);
+          done();
+        }).catch(done);
+      });
+
+    });
+
+  });
+
+  describe('.getLatestVersion()', function() {
+
+    describe('given a valid production ETCHER_FAKE_S3_LATEST_VERSION environment variable', function() {
+
+      beforeEach(function() {
+        process.env.ETCHER_FAKE_S3_LATEST_VERSION = '9.9.9';
+      });
+
+      afterEach(function() {
+        Reflect.deleteProperty(process.env, 'ETCHER_FAKE_S3_LATEST_VERSION');
+      });
+
+      describe('given a production release type', function() {
+
+        beforeEach(function() {
+          this.getRemoteVersionsStub = m.sinon.stub(s3Packages, 'getRemoteVersions');
```

Comment:
I don't think you should be stubbing `.getRemoteVersions` when testing `ETCHER_FAKE_S3_LATEST_VERSION` ?